### PR TITLE
Created ForbiddenPythonSyntaxChecker and its associated test suite

### PR DESCRIPTION
**Pull request by sushimon**
## Motivation and Context

Certain Python elements may be disallowed for use on assessments so this checker is being created to report on them.

## Your Changes

**Description**:
- Created `ForbiddenPythonSyntaxChecker` which reports on any use of disallowed Python elements (e.g. `for` loops aren't being allowed for this assignment)
- Created test suite to check if the checker correctly reports (or ignores) Python syntax that has been disallowed (allowed)
- Created an example for this checker
- Updated the PyTA documentation to document this new checker
- Updated the CHANGELOG.md file to include this checker

**Type of change** (select all that apply):

- [x] New feature (non-breaking change which adds functionality)

## Testing

- Ran the test suite (along with pre-existing tests) in the command line and verified that they've passed
- Ran the following commands in Python to connect `pyta` with the new checker and tested it on the example for this checker
```Python
>>> import python_ta
>>> python_ta.check_all(
    module_name="examples/custom_checkers/e9950_forbidden_python_syntax.py", 
    config={"disallowed-python-syntax": ["break", "continue", "comprehension", "for", "while"]}
)  # also ran the same command except with no configs
```
- Generated a copy of the PyTA documentation and verified my checker has now been documented
- Previewed the CHANGELOG.md file to see that it now includes info about this new checker

## Questions and Comments (if applicable)

- [Feb 17]: Changes have been made to address the comments!

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have verified that the CI tests have passed. 
- [x] I have reviewed the test coverage changes reported on Coveralls. 
- [x] I have added tests for my changes. 
- [x] I have updated the CHANGELOG.md file.

### CODE REVIEW

File: python_ta/checkers/forbidden_python_syntax_checker.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        ),
+    )
+
+    # this is important so that your checker is executed before others
+    priority = -1
+
+    @only_required_for_messages("forbidden-python-syntax")
+    def visit_default(self, node: nodes.NodeNG) -> None:
+        """Visit a node in the AST."""
+        name = _pascal_case_to_lower(node.__class__.__name__)
```
Comment (david-yz-liu) [inline python_ta/checkers/forbidden_python_syntax_checker.py:48]: So the main change here is I want you to just use the name of the astroid node class (i.e., `name = node.__class__.__name__` to do the comparison. That simplifies the code a little bit too.

**Review by david-yz-liu — COMMENTED** [review]
@sushimon great work! I just left a couple of comments

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,15 +7,23 @@ and adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## Unreleased
 
-## [2.4.2] - 2023-1-31
+## [2.4.3] - 2023-02-03
```
[the remark this answers is not in the document]
Comment (sushimon) [reply CHANGELOG.md]: Fixed!

File: docs/checkers/index.md [hunk 3]
```diff
@@ -2352,6 +2352,31 @@ def is_equal_to_one(a: int) -> bool:
...
+Python syntax that shouldn't be used because it defeats the purpose of an assessment. Used for
+teaching purposes.
+
+Example:
+
+```{literalinclude} /../examples/custom_checkers/e9950_forbidden_python_syntax.py
+
+```
+
+Any disallowed syntax must be listed in the options for this checker. For example, run the following
```
[the remark this answers is not in the document]
Comment (sushimon) [reply docs/checkers/index.md]: Fixed!

File: python_ta/checkers/forbidden_python_syntax_checker.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+from astroid import nodes
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import only_required_for_messages
+from pylint.lint import PyLinter
+
+
+class ForbiddenPythonSyntaxChecker(BaseChecker):
+    """A checker class to report on the disallowed use of various Python syntax.
+
+    Includes optional checking for the following syntax:
```
[the remark this answers is not in the document]
Comment (sushimon) [reply python_ta/checkers/forbidden_python_syntax_checker.py]: Fixed!

File: python_ta/checkers/forbidden_python_syntax_checker.py (hunk 1 above)
Comment (sushimon) [reply python_ta/checkers/forbidden_python_syntax_checker.py]: Fixed!

**Review by david-yz-liu — APPROVED** [review]

**Pull request merged**